magento/pwa-studio
# [bugfix] Use formatMessage ids and default messages per api

## Description

`formatMessage(obj)` takes an object like `{ id, defaultMessage, description }`. We were passing `defaultMessage` as `id`, which "worked" but only coincidentally. If you have some variable replacement within an `id`, there is a moment where the id will render as text before being replaced by the variable. See the below video.

![](https://i.gyazo.com/b57303dd04eda11356a90804d9b71b0c.gif)

Note: the issue from account chip is being fixed in #2672 

## Related Issue





Closes [PWA-913](https://jira.corp.magento.com/browse/PWA-913).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Check all the areas that are changed and ensure they still render the right messages.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
